pkg/cmd/project/create/create.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		return nil, err
+	}
+
+	nestedOpts = append(nestedOpts, opts)
+	return nestedOpts, nil
+}
+
+func AskProjectGroups(ask question.Asker, value string, getAllGroupsCallback GetAllGroupsCallback, createProjectGroupCallback CreateProjectGroupCallback) (string, cmd.Dependable, error) {
+	if value != "" {
+		return value, nil, nil
+	}
+	var shouldCreateNewProjectGroup bool
+	ask(&survey.Confirm{
+		Message: "Would you like to create a new Project Group?",
+		Default: false,
+	}, &shouldCreateNewProjectGroup)
+
+	if shouldCreateNewProjectGroup {
+		return createProjectGroupCallback()
+	}
+
+	g, err := selectors.Select(ask, "You have not specified a Project group for this project. Please select one:", getAllGroupsCallback, func(pg *projectgroups.ProjectGroup) string {
+		return pg.Name
+	})
+	if err != nil {
+		return "", nil, err
+	}
+	return g.Name, nil, nil
+
+}
+
+func PromptForConfigAsCode(opts *CreateOptions) error {
+	if !opts.ConfigAsCode.Value {
+		opts.Ask(&survey.Confirm{
+			Message: "Would you like to use Config as Code?",
+			Default: false,
+		}, &opts.ConfigAsCode.Value)
+	}
+
+	if opts.ConfigAsCode.Value {
+		if opts.GitStorage.Value == "" {
+			selectedOption, err := selectors.SelectOptions[string](opts.Ask, "Select where to store the Git credentials", getGitStorageOptions)
+
+			if err != nil {
+				return err
+			}
+			opts.GitStorage.Value = selectedOption.Value
+		}
+
+		if opts.GitUrl.Value == "" {
+			if err := opts.Ask(&survey.Input{
+				Message: "Git URL",
+				Help:    "The URL of the Git repository to store configuration.",
+			}, &opts.GitUrl.Value, survey.WithValidator(survey.ComposeValidators(
+				survey.MaxLength(200),
+				survey.MinLength(1),
+				survey.Required,
+			))); err != nil {
+				return err
+			}
+		}
+
+		if opts.GitBasePath.Value == "" {
+			if err := opts.Ask(&survey.Input{
+				Message: "Git repository base path",
+				Help:    "The path in the repository where Config As Code settings are stored. Default value: '" + DefaultBasePath + "'",
+			}, &opts.GitBasePath.Value, survey.WithValidator(survey.ComposeValidators(
+				survey.MaxLength(200),
+			))); err != nil {
+				return err
+			}
+		}
+
+		if opts.GitBranch.Value == "" {
+			if err := opts.Ask(&survey.Input{
+				Message: "Git branch",
+				Help:    "The default branch to use. Default value: '" + DefaultBranch + "'",
+			}, &opts.GitBranch.Value, survey.WithValidator(survey.ComposeValidators(
+				survey.MaxLength(200),
+			))); err != nil {
+				return err
+			}
+		}
+
+		if opts.GitInitialCommitMessage.Value == "" {
+			if err := opts.Ask(&survey.Input{
+				Message: "Initial Git commit message",
+				Help:    "The commit message used in initializing. Default value: '" + DefaultGitCommitMessage + "'",
+			}, &opts.GitInitialCommitMessage.Value, survey.WithValidator(survey.ComposeValidators(
+				survey.MaxLength(50),
+			))); err != nil {
+				return err
+			}
+		}
+
+		if opts.GitStorage.Value == GitStorageLibrary {
```

Comment:
is `GitStorageLibrary` a real thing that someone might input? If so, this would be friendlier if you just returned a normal go error rather than panic

Reply:
this was a work in progress waiting on git credential support in `go-octopusdeploy`